src/components/Button.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useEffect, useRef, useState } from "react";
+import { Plus } from "@phosphor-icons/react";
+
+interface ButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  onClick?: () => void;
+  icon?: boolean;
+  text?: string;
+  rounded?: boolean;
+  fill?: boolean;
+  status?: boolean;
+}
+
+const Button: React.FC<ButtonProps> = ({
+  text = null,
+  icon = false,
+  onClick = () => console.log("Clicked"),
+  rounded = false,
+  fill = true,
+  status = true,
+}) => {
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const [fontSize, setFontSize] = useState("1em");
+
+  useEffect(() => {
```

Comment:
No need to use side effects here. Just use relative unit such as "em" to add padding to your button and size your button text. For example, I have div with text-display, then in then button if I have 2em that would be 128px.